### ISSUE TITLE
CORS Issue - Annyone else getting this?

### ISSUE THREAD

**Issue by Skylude** [gdi2290/angular-intercom#30]
XMLHttpRequest cannot load https://api-ping.intercom.io/vjs/users/ping. No 'Access-Control-Allow-Origin' header is present on the requested resource.

This is when using the script from inside angular instead of putting the script directly on the page.

**Comment by yjwong** [comment]
Seems to be ok for me. Does the issue still happen for you? The Intercom ping endpoint always returns `Access-Control-Allow-Origin: *`, so there's no reason why it would be broken.

**Comment by Skylude** [comment]
Yeah it's still happening. I'll do some more digging tomorrow. If I include the script normally in html it works but when I use async loading it throws that error. Thanks for getting back to me!

**Comment by Skylude** [comment]
Just a quick update the problem was I was sending a parameter called "companies" as a javascript list instead of company. For some reason this threw a CORS error and NOT a typical 500 response. One other question I have for you is secure mode found here, https://docs.intercom.io/configuring-intercom/enable-secure-mode

Do you plan on supporting this?

Thanks again!

Date: Sun, 28 Feb 2016 10:02:49 -0800

**Comment by yjwong** [comment]
That's interesting - CORS is actually supported, you just have to send the user_hash as part of the user options.

    $intercom.boot({
        'user_hash': ...
        ... // Rest of user properties
    });

You can generate the user_hash on the server-side.
